packages/hardhat-foundry/src/index.ts

```diff
@@ -0,0 +1,179 @@
+  }
+
+  // Set sources path
+  config.paths.sources = foundryConfig.src;
+
+  // Change hardhat's cache path if it clashes with foundry's
+  if (foundryConfig.cache_path === "cache") {
+    config.paths.cache = "cache_hardhat";
+  }
+
+  pluginActivated = true;
+});
+
+// Task that translates import names to sourcenames using remappings
+internalTask(TASK_COMPILE_TRANSLATE_IMPORT_NAME).setAction(
+  async (
+    { importName }: { importName: string },
+    _hre,
+    runSuper
+  ): Promise<string> => {
+    if (!pluginActivated) {
+      return runSuper({ importName });
+    }
+
+    const remappings = getRemappings();
+
+    for (const from in remappings) {
+      if (Object.prototype.hasOwnProperty.call(remappings, from)) {
+        const to = remappings[from];
+        if (importName.startsWith(from)) {
+          return importName.replace(from, to);
+        }
+      }
+    }
+
+    return importName;
+  }
+);
+
+// Task that includes the remappings in solc input
+internalTask(TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE).setAction(
+  async (
+    {
+      dependencyGraph,
+      file,
+    }: {
+      dependencyGraph: DependencyGraph;
+      file: ResolvedFile;
+      solidityFilesCache?: SolidityFilesCache;
+    },
+    hre,
+    runSuper
+  ): Promise<CompilationJob | CompilationJobCreationError> => {
+    const job = (await runSuper({ dependencyGraph, file })) as
+      | CompilationJob
+      | CompilationJobCreationError;
+
+    if (!pluginActivated || isCompilationJobCreationError(job)) {
+      return job;
+    }
+
+    const remappings = getRemappings();
+    job.getSolcConfig().settings.remappings = Object.entries(remappings).map(
+      (fromTo) => fromTo.join("=")
+    );
+
+    return job;
+  }
+);
+
+task(
+  TASK_INIT_FOUNDRY,
+  "Initialize foundry setup in current hardhat project",
+  async (_, hre: HardhatRuntimeEnvironment) => {
+    if (existsSync("foundry.toml")) {
+      console.log(`File foundry.toml already exists. Aborting.`);
+      return;
+    }
+
+    console.log(`Creating foundry.toml file...`);
+
+    writeFileSync(
+      "foundry.toml",
+      [
+        `[profile.default]`,
+        `src = '${path.relative(
+          hre.config.paths.root,
+          hre.config.paths.sources
+        )}'`,
+        `out = 'out'`,
+        `libs = ['node_modules', 'lib']`,
+        `test = '${path.relative(
+          hre.config.paths.root,
+          hre.config.paths.tests
+        )}'`,
+        `cache_path  = 'cache_forge'`,
+      ].join("\n")
+    );
+
+    const cmd = `forge install --no-commit foundry-rs/forge-std`;
+    console.log(`Running ${chalk.blue(cmd)}`);
+    try {
+      runCmdSync(cmd);
+    } catch (error) {
+      console.log(
+        chalk.red(
+          `Command failed. Please continue forge-std installation manually.`
+        )
+      );
+      console.log(error);
+    }
+  }
+);
+
+function isCompilationJobCreationError(
+  x: CompilationJob | CompilationJobCreationError | SolcConfig
```

Comment:
The `SolcConfig` type is not necessary here. (I know it's included in the same function from core, but here it's not used.)